app/views/devise/registrations/edit.html.erb

```diff
@@ -38,6 +38,21 @@
 
 <h3>Cancel my account</h3>
 
-<p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
+<% if resource.class.require_password_to_destroy %>
+  <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { name: 'require_password_to_destroy', method: :delete }) do |f| %>
```

Comment:
We still have problems with `non-unique id` pointed out by @tegon [here](https://github.com/plataformatec/devise/pull/4840/files#r238831376) . I think we don't need to change the `name` attribute, we should change the `id` instead: `html: { id: 'require_password_to_destroy', method: :delete }`, it should fix this problem :).